### PR TITLE
added test case for dsolver on lienard ode

### PR DESCRIPTION
**Pull request by hyadav2k**
Added test case for lienard ode solution obtained using dsolver, w.r.t the discussions on the issue.
Credits: @oscarbenjamin 
Add test example from #4414

NO ENTRY

### CODE REVIEW

File: sympy/solvers/ode/tests/test_ode.py [hunk 1]
```diff
@@ -2211,3 +2211,11 @@ def test_issue_17322():
     sol = [Eq(f(x), C1), Eq(f(x), C1*exp(-x))]
     assert set(sol) == set(dsolve(eq, hint='lie_group'))
     assert checkodesol(eq, sol) == 2*[(True, 0)]
+
+
+def test_issue_4414():
```
Comment (0sidharth) [inline sympy/solvers/ode/tests/test_ode.py:2216]: I think it is better to add this test near the closest test number, rather than at the end of the file

Comment (oscarbenjamin) [reply sympy/solvers/ode/tests/test_ode.py:2216]: The test should be added to the refactored tests in `test_single.py` along with the other cases for `2nd_linear_bessel`.

A link to the issue can be included in a comment.

**Comment by hyadav2k** [comment]
@oscarbenjamin I made the changes as specified.
Kindly suggest if further betterment can be done.
Thanks.

File: sympy/solvers/ode/tests/test_single.py [hunk 2]
```diff
@@ -1655,6 +1655,11 @@ def _get_examples_ode_sol_2nd_linear_bessel():
         'eq': (x-2)**2*(f(x).diff(x, 2)) - (x-2)*f(x).diff(x) + 4*(x-2)**2*f(x),
         'sol': [Eq(f(x), (x - 2)*(C1*besselj(1, 2*x - 4) + C2*bessely(1, 2*x - 4)))],
     },
+    #https://github.com/sympy/sympy/issues/4414
+    '2nd_lin_bessel_11': {
+        'eq': f(x).diff(x, x)+2/x*f(x).diff(x)+f(x),
+        'sol': [Eq(f(x), (C1*besselj('1/2', x) + C2*bessely('1/2', x))/sqrt(x))],
```
Comment (oscarbenjamin) [inline sympy/solvers/ode/tests/test_single.py:1661]: There should be a blank line above the comment (and a space between `#` and `h`).

The equation would be easier to read with spaces before and after `+`.

We should not depend on string input to functions so use `S(1)/2` like the tests above.

Comment (hyadav2k) [reply sympy/solvers/ode/tests/test_single.py]: Sorry for the trouble, I made the changes as specified.
Thanks

**Comment by oscarbenjamin** [comment]
Looks good. Thanks!

**Comment by oscarbenjamin** [comment]
I edited the release note so that it doesn't say that it "fixes" the issue because the other part of the issue is about solving Lienard type ODEs.

**Comment by hyadav2k** [comment]
> I edited the release note so that it doesn't say that it "fixes" the issue because the other part of the issue is about solving Lienard type ODEs.

Sure!

**Pull request merged**